packages/storage-plus/src/prefix.rs

```diff
@@ -3,10 +3,22 @@ use serde::de::DeserializeOwned;
 use serde::Serialize;
 use std::marker::PhantomData;
 
-use crate::helpers::nested_namespaces_with_key;
-use crate::iter_helpers::{deserialize_kv, range_with_prefix};
 use cosmwasm_std::{Order, StdResult, Storage, KV};
 
+use crate::helpers::nested_namespaces_with_key;
+use crate::iter_helpers::{concat, deserialize_kv, trim};
+
+/// Bound is used to defines the two ends of a range, more explicit than Option<u8>
+/// None means that we don't limit that side of the range at all.
+/// Include means we use the given bytes as a limit and *include* anything at that exact key
+/// Exclude means we use the given bytes as a limit and *exclude* anything at that exact key
+#[derive(Copy, Clone, Debug)]
+pub enum Bound<'a> {
+    Include(&'a [u8]),
+    Exclude(&'a [u8]),
```

Comment:
Maybe `Inclusive` / `Exclusive` (like in inclusive / exclusive bounds) are better names here.